src/Symfony/Component/Intl/Tests/LanguagesTest.php

```diff
@@ -952,40 +1776,37 @@ public function testAlpha3CodeExists()
      */
     public function testGetAlpha3Name($displayLocale)
     {
-        $names = Languages::getNames($displayLocale);
+        $names = Languages::getAlpha3Names($displayLocale);
 
-        foreach ($names as $alpha2 => $name) {
-            $alpha3 = self::$alpha2ToAlpha3[$alpha2] ?? false;
-            if ($alpha3) {
-                $this->assertSame($name, Languages::getAlpha3Name($alpha3, $displayLocale));
-            }
+        foreach ($names as $language => $name) {
+            $this->assertSame($name, Languages::getAlpha3Name($language, $displayLocale));
         }
     }
 
-    public function testGetAlpha3NameWithInvalidCountryCode()
+    public function testGetAlpha3NameWithInvalidLanguageCode()
     {
         $this->expectException(MissingResourceException::class);
 
-        Languages::getAlpha3Name('ZZZ');
+        Languages::getAlpha3Name('zzz');
     }
 
     /**
      * @dataProvider provideLocales
      */
     public function testGetAlpha3Names($displayLocale)
     {
-        $names = Languages::getAlpha3Names($displayLocale);
+        $languages = array_keys($names = Languages::getAlpha3Names($displayLocale));
 
-        $alpha3Codes = array_keys($names);
-        sort($alpha3Codes);
-        $this->assertSame(array_values(self::$alpha2ToAlpha3), $alpha3Codes);
+        sort($languages);
+
+        $this->assertNotEmpty($languages);
+        $this->assertEmpty(array_diff($languages, self::$alpha3Codes));
 
-        $alpha2Names = Languages::getNames($displayLocale);
-        foreach ($alpha2Names as $alpha2Code => $name) {
-            if (!isset(self::$alpha2ToAlpha3[$alpha2Code])) {
-                unset($alpha2Names[$alpha2Code]);
+        foreach (Languages::getNames($displayLocale) as $alpha2Code => $name) {
+            $alpha3Code = self::$alpha2ToAlpha3[$alpha2Code] ?? (3 === \strlen($alpha2Code) ? $alpha2Code : null);
+            if (null !== $alpha3Code) {
+                $this->assertSame($name, $names[$alpha3Code]);
```

Comment:
Ok, at least you check that the translations that are there are the same.

It is not the strong test that I wanted, but I guess we have to agree to disagree on that one.

Reply:
well for now we cant make it any stronger, not until we proof the lists are of the same size.